### PR TITLE
Merge `source_map_stack_trace` package

### PR DESCRIPTION
**Pull request by mosuem**
Import `package:source_map_stack_trace`

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: .github/workflows/source_map_stack_trace.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        working-directory: pkgs/source_map_stack_trace/
+    strategy:
+      fail-fast: false
+      matrix:
+        # Add macos-latest and/or windows-latest if relevant for this package.
+        os: [ubuntu-latest]
+        sdk: [3.3.0, dev]
```
Comment (devoncarew) [inline .github/workflows/source_map_stack_trace.yml:59]:
```suggestion
        sdk: [3.3, dev]
```

File: pkgs/source_map_stack_trace/CONTRIBUTING.md
Comment (devoncarew) [inline pkgs/source_map_stack_trace/CONTRIBUTING.md:1]: We could delete this file, here or in a follow-up; I doubt it has useful diffs from the repo / dart-lang/.github contributing file.

File: pkgs/source_map_stack_trace/LICENSE [hunk 2]
```diff
@@ -0,0 +1,27 @@
+Copyright 2015, the Dart project authors.
```
Comment (devoncarew) [inline pkgs/source_map_stack_trace/LICENSE:1]:
```suggestion
Copyright 2015, the Dart project authors.
```

File: pkgs/source_map_stack_trace/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: source_map_stack_trace
+version: 2.1.2-wip
+description: A package for applying source maps to stack traces.
+repository: https://github.com/dart-lang/tools/tree/main/pkgs/source_map_stack_trace
```
Comment (devoncarew) [inline pkgs/source_map_stack_trace/pubspec.yaml:4]: 👍

**Review by devoncarew — APPROVED** [review]
🎉 !

**Comment by mosuem** [comment]
@devoncarew Could you enable non-squash merging on this repo?

**Pull request merged**